sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -515,17 +508,31 @@ class GpuHiveDelimitedTextPartitionReader(conf: Configuration,
     // Given that Table.readCsv presents the output columns in the order of the input file,
     // we need to reorder the table read from the input file in the order specified in
     // [[requestedOutputDataSchema]] (i.e. requiredAttributes).
-    val table = super.readToTable(dataBuffer, dataSize, inputFileCudfSchema,
-                                  requestedOutputDataSchema, isFirstChunk)
-    withResource(table) { table =>
-      val requiredColumnSequence = requestedOutputDataSchema.map(_.name).toList
-      val requiredColumnSet      = requiredColumnSequence.toSet
-      val outputColumnNames      = inputFileCudfSchema.getColumnNames
-                                                      .filter(c => requiredColumnSet.contains(c))
-      val reorderedColumns       = requiredColumnSequence.map(
-        colName => table.getColumn(outputColumnNames.indexOf(colName)))
-
-      new Table(reorderedColumns: _*)
+
+    withResource(splitTable) { _ =>
+      val nullFormat = conf.get("serialization.null.format", "\\N")
+      withResource(Scalar.fromString(nullFormat)) { nullTag =>
+        withResource(Scalar.fromNull(DType.STRING)) { nullVal =>
+          // This is a bit different because we are dropping columns/etc ourselves
+          val requiredColumnSequence = requestedOutputDataSchema.map(_.name).toList
+          val outputColumnNames = inputFileCudfSchema.getColumnNames
+          val reorderedColumns = requiredColumnSequence.safeMap { colName =>
+            val colIndex = outputColumnNames.indexOf(colName)
+            if (splitTable.getNumberOfColumns > colIndex) {
+              val col = splitTable.getColumn(colIndex)
+              withResource(col.equalTo(nullTag)) { shouldBeNull =>
+                shouldBeNull.ifElse(nullVal, col)
+              }
+            } else {
+              // the column didn't exist in the output, so we need to make an all null one
+              ColumnVector.fromScalar(nullVal, splitTable.getRowCount.toInt)
```

Comment:
I think I've missed something. Under what scenario might a column be in the `requestedOutputDataSchema`, but not read from the file?
Is it for a partitioned table where columns might have been dropped?

Reply:
The split has no knowledge of requestedOutputDataSchema. So if the entire file is missing columns at the end, it will not create columns for them in the table it outputs. This is here for that situation.